src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprAllRecipes.java

```diff
@@ -0,0 +1,131 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.RecipeUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.Recipe;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+@Name("All Recipes")
+@Description({
+	"Retrieve all recipes registered in the server",
+	"<ul>",
+	"<li>You can retrieve all recipes of a recipe type.</li>",
+	"<li>You can retrieve all recipes of an item, including custom items.</li>",
+	"<li>You can retrieve all minecraft recipes.</li>",
+	"<li>You can retrieve all custom recipes made by any and all plugins.</li>",
```

Comment:
this is pretty awkward. Don't repeat the starting half of the sentence every time. Or better yet, write a paragraph explaining it.